sys/net/gnrc/conn/gnrc_conn.c

```diff
@@ -38,7 +75,7 @@ int gnrc_conn_recvfrom(conn_t *conn, void *data, size_t max_len, void *addr, siz
                 }
                 LL_SEARCH_SCALAR(pkt, l3hdr, type, conn->l3_type);
                 if (l3hdr == NULL) {
-                    msg_send_to_self(&msg); /* requeue invalid messages */
+                    msg_send_to_self(&msg); /* requeue invalid messages, wouldn't this make the thread just read the message 3 times and return? */
```

Comment:
@DipSwitch, I had problems understanding this concept in the beginning, too. The idea is that this IPC call will pop the message from the head of the queue and enqueue it at the end.